scipy/_lib/_util.py

```diff
@@ -32,6 +32,15 @@
         AxisError, ComplexWarning, VisibleDeprecationWarning  # noqa: F401
     )
 
+np_int: type
+np_uint: type
+
+if np.lib.NumpyVersion(np.__version__) >= "2.0.0":
```

Comment:
This will fail on dev versions, so should be a bit more complex:
```python
if np.lib.NumpyVersion(np.__version__) >= "2.0.0.dev0":
    try:
        np_int = np.long
        np_uint = np.ulong
    except AttributeError:
            np_int = np.int_
            np_uint = np.uint
else:
    np_int = np.int_
    np_uint = np.uint
```

Would it make sense to use `np_long` here as the name? Seems a bit more understandable, since this is a C long.